### PR TITLE
Add a definition for 64-bit unsigned ints

### LINKED ISSUE #4289: The format of "64-bit unsigned integers" is never specified
Section [Timestamp Query](https://gpuweb.github.io/gpuweb/#timestamp) says that `resolveQuerySet` writes timestamps to the buffer as "64-bit unsigned integers". Usually the WebGPU spec defers to the WGSL spec to spell out the byte-level representations of values, but since WGSL doesn't have a `u64` type at the moment this happens not to be covered.

We could simply change Timestamp Query to say "little-endian", but we've avoided the use of "endian" so far. That seems like a good thing, and we should keep it up.

We could add a section on `u64` to the WGSL spec, with a note explaining that it's only used in the WebGPU API. That would be most consistent with what we're doing elsewhere.

### PR DESCRIPTION
**Pull request by toji**
Fixes #4289

Following guidance in the linked issue, I've added the definition and description of the byte layout to the WGSL spec with a note that it's not a type that can be used by WGSL and then referenced it in the WebGPU spec. If there's anything we should change here to further clarify the relationship, I'm happy to do it!

### CODE REVIEW

File: wgsl/index.bs
Comment (toji) [reply wgsl/index.bs:10236]: Oh, ha! I didn't even think about that, I copied a different reference to the spec. So TIL too!

**Comment by toji** [comment]
Thanks Kai! I'll wait for one of the WGSL editors to approve before merging.

**Comment by toji** [comment]
Review ping from @dneto0 or @alan-baker?

**Comment by alan-baker** [comment]
We discussed in the WGSL editors' meeting 2024-05-29. @dneto to review.

File: wgsl/index.bs
Comment (dneto0) [inline wgsl/index.bs:2672]: I'm unhappy with how this names `u64` just like the other integer types.

"Note"s aren't supposed to be normative. 
I would rather do something like what's done for f64 in the [Floating Point Conversion](https://gpuweb.github.io/gpuweb/wgsl/#floating-point-conversion) section:

> A hypothetical type corresponding to a binary format defined by the [IEEE-754](https://gpuweb.github.io/gpuweb/wgsl/#biblio-ieee-754) floating point standard.

I suggest removing this section, and modifying the later block...

File: wgsl/index.bs
Comment (dneto0) [inline wgsl/index.bs:10225]: > &lt;dfn&gt;64-bit integer layout:&lt;/dfn&gt; Some features of the WebGPU API write 64-bit unsigned integer values into buffers.  When such a value |V| appears at byte offset |k| of a host-shared buffer, then:

...

> Note: WGSL does not have a [=concrete=] 64-bit integer type.


Then the WebGPU API should link to "64-bit integer layout"

**Review by dneto0 — CHANGES_REQUESTED** [review]
Thanks for your patience.
I've made suggestions to improve this.

File: wgsl/index.bs
Comment (toji) [reply wgsl/index.bs]: Makes sense. Done!

File: wgsl/index.bs
Comment (toji) [reply wgsl/index.bs]: SGTM!

**Review by dneto0 — APPROVED** [review]
Thank you!

**Pull request merged**
